sys/auto_init/saul/auto_init_io1_xplained.c

```diff
@@ -72,11 +72,14 @@ void auto_init_io1_xplained(void)
         saul_reg_add(&(saul_entries[i * 4]));
 
         /* GPIOs */
-        for (unsigned j = 1; j < 4; j++) {
-            saul_entries[i * 4 + j].dev = &(io1_xplained_saul_gpios[j]);
-            saul_entries[i * 4 + j].name = io1_xplained_saul_info[i][j].name;
-            saul_entries[i * 4 + j].driver = &gpio_out_saul_driver;
-            saul_reg_add(&(saul_entries[i * 4 + j]));
+        for (unsigned j = 0;
+             j < sizeof(io1_xplained_saul_gpios) / sizeof(io1_xplained_saul_gpios[0]);
+             j++) {
+            saul_reg_t *entry = &saul_entries[i * 4 + j + 1];
+            entry->dev = &(io1_xplained_saul_gpios[j]);
+            entry->name = io1_xplained_saul_info[i][j].name;
```

Comment:
Actually here it still needs to be `j + 1`, since `io1_xplained_saul_info[i][0]` is used [above](https://github.com/RIOT-OS/RIOT/pull/8960/files#diff-066b63873485a4b7b29b5b328a53e1a9R70).

Reply:
You are right. Finally starting to loop from 1 also has advantages.

I don't understand why it doesn't work now. I'll re-test again (but I have to walk to @rfuentess's office to grab the board...)

Reply:
Just tested again and all LEDs work as expected:
```
# Power on extension LED
> saul write 3 1
# Power off extension LED
> saul write 3 0
# Plug a LED on GPIO2 extension:
> saul write 5 0
> saul write 5 1
```
All is working. Are you sure the extention's LED is not working ? Can you tell me how you wire the 2 GPIOs of the extentions (you need to put one side on one of the gpio1/2 pin and the other side on the GND pin, under VTG) ?

Reply:
m) I think used the wrong saul ID (0) for writing. Tested now all pins and it worked!

Reply:
Sorry for the hassle :-/ (though you should test before pushing ;-))